src/Umbraco.Web.UI.Client/src/views/propertyeditors/blocklist/umbBlockListPropertyEditor.component.js

```diff
@@ -220,8 +220,13 @@
                 });
             }
 
-            // TODO: Why is there a '/' prefixed? that means this will never work with virtual directories
-            block.view = (block.config.view ? "/" + block.config.view : getDefaultViewForBlock(block));
+            if(block.config.view) {
+                block.config.view = umbRequestHelper.ensureVirtualPath(block.config.view);
```

Comment:
This shouldn't need to be done in 2 calls, IMO the convertVirtualToAbsolutePath should implicitly call ensureVirtualPath so there's no need to explicitly call that.